### ISSUE TITLE
RainMachine - zone control issue

### ISSUE THREAD

**Issue by Nbr1Sniper** [home-assistant/home-assistant.io#18074]
### Feedback

RainMachine integration zone control currently allows for multiple zones to be on at once which means there isn’t enough pressure to run either zone. It should turn off the zone that was on just before the next zone is selected and turned on manually. This would follow how the native app works which the HA integration should follow. I don’t think I should need to turn off a zone first in order to turn on the next manually. 

I understand why it does this because it’s a basic switch but I think it should have an off of other switches when a zone is switched on

### URL

https://www.home-assistant.io/integrations/rainmachine/

### Version

2021.6.0

**Comment by bachya** [comment]
@Nbr1Sniper I want to confirm: you're saying that since RainMachine itself can't have multiple zones running at the same time, the HASS integration shouldn't, either (i.e., when a zone is turned on, it should first turn off all other zones)?

**Comment by Nbr1Sniper** [comment]
> @Nbr1Sniper I want to confirm: you're saying that since RainMachine itself can't have multiple zones running at the same time, the HASS integration shouldn't, either (i.e., when a zone is turned on, it should first turn off all other zones)?

Yes that is correct. I thought about creating a rule within Node Red to do this but don’t think the integration should allow it to happen in general. Don’t want a newbie (like me) to mess up their system because it overrides what the native app does. Appreciate all you do!

**Comment by bachya** [comment]
@Nbr1Sniper Got it! I'll get working on a fix (hopefully in time for 2021.6.1). Thanks!

**Comment by Nbr1Sniper** [comment]
> @Nbr1Sniper Got it! I'll get working on a fix (hopefully in time for 2021.6.1). Thanks!

Amazing!!!!!!!!!!!!  Thank you!!!

**Comment by bachya** [comment]
@Nbr1Sniper After digging in, I actually think the existing functionality is already representative of what the RainMachine controller does.

I have two zones at my house: `Landscaping` and `Flower Box`. As we already know, I can turn both zones on simultaneously within HASS:

![CleanShot 2021-06-03 at 10 03 27](https://user-images.githubusercontent.com/47216/120676134-fb745680-c452-11eb-97c1-7238ca33e85d.png)

What this tactically does in the controller is activate the first zone and queue the second – so, if turn on `Landscaping` first, the controller shows `Landscaping` as running and `Flower Box` as queued:

![CleanShot 2021-06-03 at 09 59 42](https://user-images.githubusercontent.com/47216/120676286-1941bb80-c453-11eb-940b-48ee02c1097c.png)

We can see this reflected in the HASS zone switches, too, via attributes – for instance, the `Landscaping` switch has this attribute:

![CleanShot 2021-06-03 at 10 07 34](https://user-images.githubusercontent.com/47216/120676725-91a87c80-c453-11eb-919c-d41b1c42e70b.png)

...and the `Flower Box` switch has this attribute:

![CleanShot 2021-06-03 at 09 59 59](https://user-images.githubusercontent.com/47216/120676501-5312c200-c453-11eb-894d-c5995d28da0b.png)

---

It seems like this representation of two "active" zones, one running and one queued, is more reflective of what's actually going on. Deactivating "active" zones (like we've been discussing) doesn't have that same reflection; plus, turning off a HASS switch would actually remove a zone from the queue, which seems incorrect.

Thoughts? Am I missing something?
